### ISSUE TITLE
@def suppresses JAVADOC_AUTOBRIEF=YES

### ISSUE THREAD

**Issue by paul-j-lucas** [doxygen/doxygen#7852]
**Describe the bug**
For a C macro definition that explicitly includes `@def`:
```
#ifndef bug_H
#define bug_H

/**
 * @def C_FALLTHROUGH
 *
 * Intentionally fall through to the next `switch` `case`.
 */
#define C_FALLTHROUGH             __attribute__((fallthrough))

#endif
```
and `JAVADOC_AUTOBRIEF` set to `YES`, the generated output does **_not_** do the auto-brief. If you remove the `@def`, it does.

**Expected behavior**
I expect that the presence or absence of `@def` should not affect auto-brief.

**To Reproduce**
Use the above file `bug.h` with this configuration file:
```
PROJECT_NAME            = "bug"

INPUT                   = bug.h
FILE_PATTERNS           = *.h
OUTPUT_DIRECTORY        = docs

DISABLE_INDEX           = YES
GENERATE_LATEX          = NO
JAVADOC_AUTOBRIEF       = YES
MARKDOWN_SUPPORT        = YES
MAX_INITIALIZER_LINES   = 0
OPTIMIZE_OUTPUT_FOR_C   = YES
```

**Version**
Doxygen 1.8.18 on macOS 10.15.5.

**Comment by doxygen** [comment]
You assumption is wrong; `@def` does end a brief description just like `@fn`, `@var`, `@typedef`, `@class` etc. My suggestion it to remove the line with `@def` as it is completely redundant.

**Comment by paul-j-lucas** [comment]
> My suggestion it to remove the line with @def as it is completely redundant.

The example I gave is a whittled down version of the real file. The real file has `C_FALLTHROUGH` inside an `#ifdef`. Later in the same file:
```
#ifndef C_FALLTHROUGH
#define C_FALLTHROUGH             ((void)0)
#endif /* C_FALLTHROUGH */
```
Without adding `@def`, Doxygen uses _that_ definition as the one for the documentation. Since that definition has no Doxygen comment, `C_FALLTHROUGH` is not documented. To fix that, I explicitly put `@def` on the actual definition earlier to force Doxygen to use the earlier one.

**Comment by doxygen** [comment]
> > My suggestion it to remove the line with @def as it is completely redundant.
> 
> The example I gave is a whittled down version of the real file. The real file has `C_FALLTHROUGH` inside an `#ifdef`. Later in the same file:
> 
> ```
> #ifndef C_FALLTHROUGH
> #define C_FALLTHROUGH             ((void)0)
> #endif /* C_FALLTHROUGH */
> ```
> 
> Without adding `@def`, Doxygen uses _that_ definition as the one for the documentation. Since that definition has no Doxygen comment, `C_FALLTHROUGH` is not documented. To fix that, I explicitly put `@def` on the actual definition earlier to force Doxygen to use the earlier one.

Interesting (but hopefully rare case).
Then I would suggest to be explicit and use:

```c
/**
 * @def C_FALLTHROUGH
 * @brief Intentionally fall through to the next `switch` `case`.
 */
#define C_FALLTHROUGH             __attribute__((fallthrough))
```

**Comment by paul-j-lucas** [comment]
> `@def` does end a brief description just like `@fn`, `@var`, `@typedef`, `@class` etc.

Then your documentation is incomplete. Nowhere in your documentation for any of those commands does it mention "brief" much less ending it.

That aside, why _should_ it end a brief description? All the `@def` should mean is "Hey Doxygen! This comment is the documentation for this thing," and nothing else. Why should it have any side-effects at all?

**Comment by paul-j-lucas** [comment]
> Then I would suggest to be explicit and use: ..

Actually, I discovered that if I do:
```
/**
 * Intentionally fall through to the next `switch` `case`.
 * @def C_FALLTHROUGH
 */
#if __has_attribute(fallthrough) || GCC_AT_LEAST_VERSION(7,0)
#define C_FALLTHROUGH             __attribute__((fallthrough))
#endif
```
i.e., put the `@def` _after_ the description, it works as I expect. As I mentioned in my previous comment, `@def` shouldn't have side-affects... only "tag" a particular comment as being _the_ documentation for a thing regardless of where within the comment one puts it.

**Comment by doxygen** [comment]
> > `@def` does end a brief description just like `@fn`, `@var`, `@typedef`, `@class` etc.
> 
> Then your documentation is incomplete. Nowhere in your documentation for any of those commands does it mention "brief" much less ending it.
> 
> That aside, why _should_ it end a brief description? All the `@def` should mean is "Hey Doxygen! This comment is the documentation for this thing," and nothing else. Why should it have any side-effects at all?

See the rationale I added to my first reaction. Maybe it should have been designed differently, but now it behaves like it does. Changing this will probably also break a lot of existing comments.

**Comment by paul-j-lucas** [comment]
Perhaps augment the documentation for `@def` to be explicit about its interaction with brief?

**Comment by albert-github** [comment]
A patch (commit: 327423e217337bf4e64515aba6cf78b9877bc165) has been integrated into the master version.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
